src/frontend/hooks/server/media.ts

```diff
@@ -1,52 +1,99 @@
 import {Observation} from '@mapeo/schema';
 import {BlobVariant} from '@mapeo/core/dist/types';
-import {useMutation, useQuery} from '@tanstack/react-query';
+import {useMutation, useQueries} from '@tanstack/react-query';
+import {SetRequired} from 'type-fest';
+import {URL} from 'react-native-url-polyfill';
 
-import {Photo} from '../../contexts/PhotoPromiseContext/types';
+import {DraftPhoto} from '../../contexts/PhotoPromiseContext/types';
 import {useProject} from './projects';
 
+type SavablePhoto = SetRequired<
+  Pick<DraftPhoto, 'originalUri' | 'previewUri' | 'thumbnailUri'>,
+  'originalUri'
+>;
+
 export function useCreateAttachmentsMutation() {
   const project = useProject();
 
   return useMutation({
-    mutationFn: async (photos: Photo[]) => {
+    mutationFn: async (photos: SavablePhoto[]) => {
       if (!project) throw new Error('Project instance does not exist');
 
       return Promise.all(
         photos.map(p => {
-          // TODO: Fix
+          const {originalUri, previewUri, thumbnailUri} = p;
+
           return project.$blobs.create(
-            {original: 'abc'},
-            {mimeType: 'img/png'},
+            {
+              original: new URL(originalUri).pathname,
+              preview: previewUri ? new URL(previewUri).pathname : undefined,
+              thumbnail: thumbnailUri
+                ? new URL(thumbnailUri).pathname
+                : undefined,
+            },
+            // TODO: DraftPhoto type should probably carry MIME type info that feeds this
+            // although backend currently only uses first part of path
+            {mimeType: 'image/jpeg'},
```

Comment:
not sure if there's ever a case where we take photos of different mime types but probably useful to include the mime type info as part of the photo type in general moving forward. better to do as a follow-up though